tests/TestHarness/launcher.py

```diff
@@ -66,9 +188,264 @@ def comma_separated(string):
+        if non_bios > 2:
+            end_of_loop = False
+            i = self.start_ndx()
+            while not end_of_loop:
+                front = i
+                front, end_of_loop = self.next_ndx(front)
+                if end_of_loop and not make_ring:
+                    break
+                self.network.nodes[self.aliases[i]].peers.append(self.aliases[front])
+                i, _ = self.next_ndx(i)
+        elif non_bios == 2:
+            n0 = self.start_ndx()
+            n1 = n0
+            n1 = self.next_ndx(n1)[0]
+            self.network.nodes[self.aliases[n0]].peers.append(self.aliases[n1])
+            if make_ring:
+                self.network.nodes[self.aliases[n1]].peers.append(self.aliases[n0])
+
+    def make_line(self, make_ring: bool = True):
+        print(f"making {'ring' if make_ring else 'line'}")
+        self.bind_nodes()
+        nl = list(self.network.nodes.values())
+        for node, nextNode in zip(nl, nl[1:]):
+            node.peers.append(nextNode.name)
+        if make_ring:
+            nl[-1].peers.append(nl[1].name)
+
+    def make_star(self):
+        print('making star')
+        non_bios = self.args.total_nodes - 1
+        if non_bios < 4:
+            self.make_line()
+            return
+        self.bind_nodes()
+
+        links = 3
+        if non_bios > 12:
+            links = int(math.sqrt(non_bios)) + 2
+        gap = 3 if non_bios > 6 else (non_bios - links)/2 + 1
+        while non_bios % gap == 0:
+            gap += 1
+
+        loop = False
+        i = self.start_ndx()
+        while not loop:
+            current = self.network.nodes[self.aliases[i]]
+            ndx = i
+            for l in range(1, links+1):
+                ndx = self.skip_ndx(ndx, l * gap)
+                peer = self.aliases[ndx]
+                found = True
+                while found:
+                    found = False
+                    for p in current.peers:
+                        if p == peer:
+                            ndx, _ = self.next_ndx(ndx)
+                            if ndx == i:
+                                ndx, _ = self.next_ndx(ndx)
+                            peer = self.aliases[ndx]
+                            found = True
+                            break
+                if peer != current.name:
+                    current.peers.append(peer)
+            i, loop = self.next_ndx(i)
+
+    def make_mesh(self):
+        print('making mesh')
+        non_bios = self.args.total_nodes - 1
+        self.bind_nodes()
+        loop = False
+        i = self.start_ndx()
+        while not loop:
+            current = self.network.nodes[self.aliases[i]]
+            for j in range(1, non_bios):
+                ndx = self.skip_ndx(i,j)
+                peer = self.aliases[ndx]
+                if peer not in current.peers:
+                    current.peers.append(peer)
+            i, loop = self.next_ndx(i)
+
+    def make_custom(self):
+        print('making custom')
```

Comment:
Why not implement `make_custom` yet?

Reply:
It was implemented using macro reflection in the C++ code, which isn't available to the Python side, and it's reportedly an extremely rarely used option.

Reply:
Is the thought to drop support for it completely and make a different network topology the default?

Reply:
Most likely yes.